.github/workflows/reusable_docker_package.yaml

```diff
@@ -0,0 +1,151 @@
+name: Reusable docker package make and upload
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        type: string
+        description: "Target os. Default value is debian-11."
+        default: 'debian-11'
+      arch:
+        type: string
+        description: "Target architecture. Default value is amd."
+        default: 'amd'
+      runner_arch_label:
+        type: string
+        description: "Runner architecture label. Amd is built on X64, arm on ARM64. Default value is X64."
+        default: 'X64'
+      toolchain:
+        type: string
+        description: "Toolchain version. Default value is v5."
+        default: 'v5'
+      build_type:
+        type: string
+        description: "Memgraph Build type. Default value is Release."
+        default: 'Release'
+      artifact_name:
+        type: string
+        description: "Unique artifact name, pass it in to override the default. Default value is docker-arch."
+        default: ''
+      additional_build_args:
+        type: string
+        description: "Additional build flags (--for-platform, --for-docker ...). Default value is empty."
+        default: ''
+      timeout_minutes:
+        type: number
+        description: "Timeout in minutes for the job. Default value is 60."
+        default: 60
+      push_to_s3:
+        type: boolean
+        description: "Should the final package be pushed to an S3 bucket. Default value is false"
+        default: false
+      s3_bucket:
+        type: string
+        description: "Name of the target S3 bucket."
+      s3_region:
+        type: string
+        description: "Region for the target S3 bucket."
+      s3_dest_dir:
+        type: string
+        description: "Target dir path in S3 bucket."
```

Comment:
Is it hypothetically possible to join all 4 of these together? It seems to open the door for mistakes having them all separate.

Reply:
Or at least validate them up front as part of a pre-flight check, please

Reply:
When running `package_memgraph.yaml` manually (as it is intended) `s3_bucket` and `s3_region` are chosen from preset valid options. `s3_dest_dir` should be checked before if we are pushing to `download.memgraph.com`, this is where official downloadable binaries are. I will add this.